module4/projects/solo_stretch_tech/index.md

```diff
@@ -3,14 +3,137 @@ title: Solo Stretch Tech
+- Two stretch technologies (1 BE and 1 FE) implemented
+- The frontend and backend should "talk" at least 3 different places in the app, with at least one request mutating the data (POST, PATCH, PUT, DELETE)
 
-## Evaluation Details
+## Stretch Technologies
 
+**We expect you to implement 1 stretch tech from each menu below (2 total).** If you would like to add more, you can, but be careful not to bite off too much. If there is a technology you're interested in using that is not on these lists, let your instructor know and we'll do our best to make it work.
+
+<section class="dropdown">
+### Frontend Menu
+
+- **Global State Management:** Use a tool like Redux or Context API to handle data management. Student should be able to speak to the benefits of using global statement management in a React application.
+- **Another Framework:** Build a frontend using a different framework like Vue, Svelte, Angular, etc. Student should identify strengths and pain points for this framework in comparison to React and vanilla JS. 
+- **Unit Testing:** In addition to using Cypress for end-to-end testing, implement unit testing in your app using React Testing Library or Jest. 
+- **Progressive Web App:** Create a PWA that (at minimum) uses caching to maintain some offline functionality and fast loading, is fully responsive to screen size, and has an app-like feel for users. 
+- **CSS Technologies:** Up your CSS game with GSAP Animations, WebGL Animations, Bootstrap, Styled Components, or something else you find interesting. If you choose this stretch tech, we'd expect your app design to be extremely strong. 
+- **Mobile Development**: Create an app that uses a frontend mobile language like React Native or Swift. 
+</section>
+
+<section class="dropdown">
+### Backend Menu
+
+- **Caching:** Use low-level caching (no FE caching). Students should be able to identify how the points where caching is implemented help to alleviate the main workflow of the application, and ideally show before/after data.
+- **Background Jobs:** Consider using Sidekiq with Redis. Students should be able to identify how the background worker(s) alleviate the main workflow of the application, and ideally show before/after data.
+- **Rate Limiting:** Consider using the gem called “rack-attack” and test to demonstrate it’s working! Students should be able to discuss why rate limiting is an effective feature. Ideally, be able to demonstrate it live using a small script (either live or in production).
+- **Microservice:** Students should create a separate microservice (separate from their BE API app) using Sinatra or a different framework, and be able to speak to the pros/cons of that framework when compared to Rails.
+- **JWT (JSON Web Tokens):** Use the jwt gem for stateless authentication alongside username/password login. Students should explain its benefits and show token expiration and verification
+<!-- **I used chatGPT for the following list items, please change as needed** -->
```

Comment:
Flagging this comment for BE eyes

Reply:
These all look like great stretches to me